### PR TITLE
feat: Include opt-in Red Hat telemetry

### LINKED ISSUE #25: Include telemetry
As first iteration, only track the number of installation, startup with OS, country and version of the extension.

it will help to monitor adoption and provide arguments to slow down or accelerate development

### PR DESCRIPTION
**Pull request by apupier**
followed documentation from
https://github.com/redhat-developer/vscode-redhat-telemetry#how-to-use-this-library

currently tracking only generic data on activation and deactivation of the extension

fixes #25

next interesting thing to track:
- opening of the Kaoto editor https://github.com/KaotoIO/vscode-kaoto/issues/55

### CODE REVIEW

**Comment by apupier** [comment]
seems VS Code telemetry module is nto ready for webpack :'-(

a lot of compile error of this kind:
```
WARNING in ./node_modules/@redhat-developer/vscode-redhat-telemetry/lib/che/cheIdManager.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/runner/work/vscode-kaoto/vscode-kaoto/vscode-kaoto/node_modules/@redhat-developer/vscode-redhat-telemetry/src/che/cheIdManager.ts' file: Error: ENOENT: no such file or directory, open '/home/runner/work/vscode-kaoto/vscode-kaoto/vscode-kaoto/node_modules/@redhat-developer/vscode-redhat-telemetry/src/che/cheIdManager.ts'
 @ ./node_modules/@redhat-developer/vscode-redhat-telemetry/lib/services/idManagerFactory.js 4:23-53
 @ ./node_modules/@redhat-developer/vscode-redhat-telemetry/lib/vscode/redhatService.js 37:[27](https://github.com/KaotoIO/vscode-kaoto/actions/runs/3272004889/jobs/5382483134#step:8:28)-66
 @ ./node_modules/@redhat-developer/vscode-redhat-telemetry/lib/index.js 4:24-57
 @ ./src/extension/extension.ts 7:0-77 47:[31](https://github.com/KaotoIO/vscode-kaoto/actions/runs/3272004889/jobs/5382483134#step:8:32)-47
 ```

locally when launching in debug it is working fine

**Comment by apupier** [comment]
last compilation issue fixed with:

https://github.com/KaotoIO/vscode-kaoto/pull/50/files#diff-1fb26bc12ac780c7ad7325730ed09fc4c2c3d757c276c3dacc44bfe20faf166fR117-R120

**Comment by apupier** [comment]
example of report that we can get in Woopra (here on the test instance)

image

File: webpack.config.js [hunk 1]
```diff
@@ -105,14 +106,18 @@ const commonConfig = (env) => {
...
         path: require.resolve("path-browserify"),
         os: require.resolve("os-browserify/browser"),
         fs: false,
         child_process: false,
         net: false,
         buffer: require.resolve("buffer/"),
+		assert: false,
+		process: false,
+		stream: false,
+		util: false
```
Comment (apupier) [inline webpack.config.js:120]: based on https://coreos.slack.com/archives/CK14U2L4V/p1666101124417369?thread_ts=1666099245.975149&cid=CK14U2L4V maybe need to change that

**Comment by apupier** [comment]
The compilation error was due to limitations on redhat-vscode-telemetry in web environment. See https://github.com/redhat-developer/vscode-redhat-telemetry/issues/21
Thiago provided a workaround to still have a way to test some parts of the extension in web mode. For web mode, there won't be telemetry and docker won't be tried to be launched (it would have failed anyway)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -74,3 +74,15 @@ The commit contains the following structural elements, to communicate intent to
...
+
+## How to launch VS Code extension during development
+
+### Desktop
+
+Inside VS Code, launch the `Run and Debug configuration` called `Run Extension`.
+
+### Web
+
+This is available for future plan. This i snot supported.
```
Comment (Delawen) [inline CONTRIBUTING.md:86]: nit-picky: typo

File: README.md [hunk 3]
```diff
@@ -12,3 +12,8 @@
 ## Features
 
 - Create and edit Kaoto (`*.kaoto.yaml` and `*.kaoto.yml`) files.
+
+## Data and Telemetry
+
+The VS Code Kaoto extension collects anonymous [usage data](USAGE_DATA.md) and sends it to Red Hat servers to help improve our products and services. Read our [privacy statement](https://developers.redhat.com/article/tool-data-collection) to learn more. This extension respects the `redhat.telemetry.enabled` setting which you can learn more about at https://github.com/redhat-developer/vscode-redhat-telemetry#how-to-disable-telemetry-reporting
+>>>>>>> 49c5b92 feat: Include opt-in Red Hat telemetry
```
Comment (kahboom) [inline README.md:19]: Did this comment sneak in here by mistake or am I seeing things?

Comment (apupier) [reply README.md:19]: mistake. I will fix it

**Pull request merged**